### PR TITLE
[MeshingApplication][MMG] Add optimization remeshing mode

### PR DESCRIPTION
**Pull request by marcnunezc**
**Description**
This PR enables the option to use MMG preserving edge lengthes, to be used to improve mesh quality, using the built-in option:

https://www.mmgtools.org/mmg-remesher-try-mmg/mmg-remesher-options/optim

This could be useful for optimization @sunethwarna @armingeiser.

If the option is active, no metric values have to be initialized or loaded into MMG, so a check is performed to skip the InitalizeMetricData().

Users can enable this by setting the strategy "Optimization" in the mmg_process.py.

**Changelog**
- Added -optim option through libcalls 
- Added "Optimization" strategy in the python process

### CODE REVIEW

**Review by loumalouomega — DISMISSED** [review]
Ok from my side

**Comment by loumalouomega** [comment]
> Ok from my side

But CI is failing (strange...)

**Comment by marcnunezc** [comment]
> > Ok from my side
> 
> But CI is failing (strange...)

It's due v5.4 which is running with the Clang build, it seems this option is not supported in MMGS in this version, just missing a check.

File: applications/MeshingApplication/python_scripts/mmg_process.py [hunk 1]
```diff
@@ -415,7 +419,7 @@ def ExecuteInitializeSolutionStep(self):
                     if self.interval.IsInInterval(current_time):
                         # We remesh if needed
                         if self.__execute_remesh():
-                            if self.strategy == "Hessian" or self.strategy == "LevelSet":
+                            if self.strategy in ["Hessian", "LevelSet", "Optimization"]:
```
Comment (philbucher) [inline applications/MeshingApplication/python_scripts/mmg_process.py]:
```suggestion
                            if self.strategy in ["Hessian", "LevelSet", "optimization"]:
```

File: applications/MeshingApplication/python_scripts/mmg_process.py [hunk 2]
```diff
@@ -375,6 +376,9 @@ def ExecuteInitialize(self):
         mmg_parameters.AddValue("debug_result_mesh",self.settings["debug_result_mesh"])
         mmg_parameters.AddValue("initialize_entities",self.settings["initialize_entities"])
         mmg_parameters.AddValue("echo_level",self.settings["echo_level"])
+        if self.strategy == "Optimization":
```
Comment (philbucher) [inline applications/MeshingApplication/python_scripts/mmg_process.py]:
```suggestion
        if self.strategy == "optimization":
```

File: applications/MeshingApplication/python_scripts/mmg_process.py (hunk 1 above)
Comment (marcnunezc) [reply applications/MeshingApplication/python_scripts/mmg_process.py]: I was trying to be consistent with the current options, but I guess that it is not a reason to keep doing things out of the style

**Review by loumalouomega — APPROVED** [review]
OK

**Pull request merged**